collie/metrics.py

```diff
@@ -49,129 +53,159 @@ def _get_user_item_pairs(user_ids: Union[np.array, torch.tensor],
+        Iterable[int] of users or items to score
+    n_items_or_users: int
+        Number of items or users in the training data
     device: string
         Device torch should use
 
     Returns
     -------
     predicted_scores: torch.tensor
-        Tensor of shape ``n_users x n_items``
+        Tensor of shape ``n_users x n_items`` or ``n_items x n_users``
 
     """
-    user, item = _get_user_item_pairs(user_ids, n_items, device)
+    user, item = _get_user_item_pairs(model, user_or_item_ids, n_items_or_users, device)
 
     with torch.no_grad():
         predicted_scores = model(user, item)
 
-    return predicted_scores.view(-1, n_items)
+    return predicted_scores.view(-1, n_items_or_users)
 
 
 def _get_labels(targets: csr_matrix,
-                user_ids: Union[np.array, torch.tensor],
+                user_or_item_ids: Union[np.array, torch.tensor],
                 preds: Union[np.array, torch.tensor],
-                device: str) -> torch.tensor:
+                device: str,
+                negative_sample_type: Literal['item', 'user'] = 'item') -> torch.tensor:
     """
-    Returns a binary array indicating which of the recommended products are in each user's target
-    set.
+    Returns a binary array indicating which of the recommended items or users are in each user's or
+    item's target set, respectively.
 
     Parameters
     ----------
     targets: scipy.sparse.csr_matrix
         Interaction matrix containing user and item IDs
-    user_ids: np.array or torch.tensor
-        Users corresponding to the recommendations in the top k predictions
+    user_or_item_ids: np.array or torch.tensor
+        Users or items corresponding to the recommendations in the top k predictions
     preds: torch.tensor
-        Top ``k`` item IDs to recommend to each user of shape (n_users x k)
+        Top ``k`` IDs to recommend to each user or item of shape (n_users_or_items x k)
     device: string
         Device torch should use
+    negative_sample_type: str
+        Type of negative sampling the Interaction matrix, ``targets``, used
 
     Returns
     -------
     labels: torch.tensor
         Tensor with the same dimensions as input ``preds``
 
     """
-    return torch.tensor(
-        (targets[user_ids[:, None], np.array(preds.detach().cpu())] > 0)
-        .astype('double')
-        .toarray(),
-        requires_grad=False,
-        device=device,
-    )
+    if negative_sample_type == 'item':
+        return torch.tensor(
+            (targets[user_or_item_ids[:, None], np.array(preds.detach().cpu())] > 0)
+            .astype('double')
+            .toarray(),
+            requires_grad=False,
+            device=device,
+        )
+    elif negative_sample_type == 'user':
+        return torch.tensor(
+            (targets[np.array(preds.detach().cpu()), user_or_item_ids[:, None]] > 0)
+            .astype('double')
+            .toarray(),
+            requires_grad=False,
+            device=device,
+        )
```

Comment:
We can reduce repeated code here a bit more: 

```suggestion
    if negative_sample_type == 'item':
        labels = targets[user_or_item_ids[:, None], np.array(preds.detach().cpu())]
    elif negative_sample_type == 'user':
        labels = targets[np.array(preds.detach().cpu()), user_or_item_ids[:, None]]

    return torch.tensor(
        (labels > 0)
        .astype('double')
        .toarray(),
        requires_grad=False,
        device=device,
    )
```